OpenRA.Mods.Common/MapFormats/DefaultMap.cs

```diff
@@ -0,0 +1,1432 @@
+			{
+				for (var x = 0; x < width; x++)
+				{
+					var uv = new MPos(x + Bounds.Left, y + top);
+
+					// FirstOrDefault will return a (MPos.Zero, Color.Transparent) if positions is empty
+					var actorColor = positions.FirstOrDefault(ap => ap.Position == uv).Color;
+					if (actorColor.A == 0)
+						terrainColor = GetTerrainColorPair(uv);
+
+					if (isRectangularIsometric)
+					{
+						// Odd rows are shifted right by 1px
+						var dx = uv.V & 1;
+						var xOffset = pxStride * (2 * x + dx);
+						if (x + dx > 0)
+						{
+							var z = y * stride + xOffset - pxStride;
+							var c = actorColor.A == 0 ? terrainColor.Left : actorColor;
+							minimapData[z++] = c.R;
+							minimapData[z++] = c.G;
+							minimapData[z++] = c.B;
+							minimapData[z] = c.A;
+						}
+
+						if (xOffset < stride)
+						{
+							var z = y * stride + xOffset;
+							var c = actorColor.A == 0 ? terrainColor.Right : actorColor;
+							minimapData[z++] = c.R;
+							minimapData[z++] = c.G;
+							minimapData[z++] = c.B;
+							minimapData[z] = c.A;
+						}
+					}
+					else
+					{
+						var z = y * stride + pxStride * x;
+						var c = actorColor.A == 0 ? terrainColor.Left : actorColor;
+						minimapData[z++] = c.R;
+						minimapData[z++] = c.G;
+						minimapData[z++] = c.B;
+						minimapData[z] = c.A;
+					}
+				}
+			}
+
+			var png = new Png(minimapData, SpriteFrameType.Rgba32, bitmapWidth, height);
+			return png.Save();
+		}
+
+		public override void Save(IReadWritePackage toPackage)
+		{
+			MapFormat = SupportedMapFormat;
+
+			var root = new List<MiniYamlNode>();
+			foreach (var field in YamlFields)
+				field.Serialize(this, root);
+
+			// HACK: map.yaml is expected to have empty lines between top-level blocks
+			for (var i = root.Count - 1; i > 0; i--)
+				root.Insert(i, new MiniYamlNode("", ""));
+
+			// Saving to a new package: copy over all the content from the map
+			if (Package != null && toPackage != Package)
+				foreach (var file in Package.Contents)
+					toPackage.Update(file, Package.GetStream(file).ReadAllBytes());
+
+			if (!LockPreview)
+				toPackage.Update("map.png", SavePreview());
+
+			// Update the package with the new map data
+			var s = root.WriteToString();
+			toPackage.Update("map.yaml", Encoding.UTF8.GetBytes(s));
+			toPackage.Update("map.bin", SaveBinaryData());
+			Package = toPackage;
+
+			// Update UID to match the newly saved data
+			Uid = ComputeUID(toPackage);
+		}
+
+		public static void UpdatePreview(ModData modData, MapPreview mp, IReadOnlyPackage p, IReadOnlyPackage parent, MapClassification classification, string[] mapCompatibility, MapGridType gridType)
+		{
+			Dictionary<string, MiniYaml> yaml;
+			using (var yamlStream = p.GetStream("map.yaml"))
+			{
+				if (yamlStream == null)
+					throw new FileNotFoundException("Required file map.yaml not present in this map");
+
+				// yaml = new MiniYaml(null, MiniYaml.FromStream(yamlStream, "map.yaml")).ToDictionary();
```

Comment:
?

Reply:
Removed comment, was a prior version of this code that I forgot to remove.